lib/bap/bap_self.ml

```diff
@@ -60,4 +60,73 @@ module Create() = struct
   let warning f = message Warning ~section:name f
   let error f = message Error ~section:name f
 
+  module Config = struct
+    let plugin_name = name
+    include Bap_config
+
+    let (/) = Filename.concat
+    let confdir = confdir / plugin_name
+    let conf_filename = confdir / "config"
+
+    exception Improper_format of string
+
+    let get_env_options () =
+      let prefix = "BAP_" ^ String.uppercase plugin_name ^ "_" in
+      let prefix_chop_key (k, v) =
+        match String.chop_prefix ~prefix k with
+        | Some k -> Some (k, v)
+        | None -> None
+      in
+      let plugin_filter_map str =
+        match String.split str ~on:'=' with
+        | [k; v] -> prefix_chop_key (k, v)
+        | _ -> None
+      in
+      Unix.environment () |>
+      Array.to_list |>
+      List.filter_map ~f:plugin_filter_map
+
+    let options () =
+      let string_splitter str =
+        match String.split str ~on:'=' with
+        | [k; v] -> (k, v)
+        | _ -> raise (Improper_format str)
+      in
+      let split_filter = List.map ~f:string_splitter in
+      let conf_file_options =
+        try
+          In_channel.with_file
+            conf_filename ~f:(fun ch -> In_channel.input_lines ch
+                                        |> split_filter)
+        with Sys_error _ -> []
+      in
+      get_env_options () |>
+      List.fold ~init:conf_file_options
+        ~f:(fun o (k, v) -> List.Assoc.add o k v)
+
+    let get name =
+      let name = String.uppercase name in
+      let search_for = options () |> List.Assoc.find in
+      search_for name
+
+    let set ~name ~data =
+      let name = String.uppercase name in
```

Comment:
Since, I suspect that I will still look sometimes at the config files directly, and I'm scared of caps locks, so can you actually stick to the lowercase instead?